src/main/java/org/jenkinsci/plugins/durabletask/BourneShellScript.java

```diff
@@ -111,16 +111,16 @@ public String getScript() {
 
         FilePath shf = c.getScriptFile(ws);
 
-        String s = script, scriptPath;
-        final Jenkins jenkins = Jenkins.getInstance();
-        if (!s.startsWith("#!") && jenkins != null) {
-            String defaultShell = jenkins.getInjector().getInstance(Shell.DescriptorImpl.class).getShellOrDefault(ws.getChannel());
-            s = "#!"+defaultShell+" -xe\n" + s;
-        }
-        shf.write(s, "UTF-8");
+        shf.write(script, "UTF-8");
         shf.chmod(0755);
```

Comment:
This should then be done conditionally: only need `a+x` if we are using a shebang.